docs/serving/samples/hello-world/helloworld-python/index.md

```diff
@@ -5,4 +5,230 @@ weight: 1
+
+   # Run the web service on container startup. Here we use the gunicorn
+   # webserver, with one worker process and 8 threads.
+   # For environments with multiple CPU cores, increase the number of workers
+   # to be equal to the cores available.
+   CMD exec gunicorn --bind :$PORT --workers 1 --threads 8 --timeout 0 app:app
+   ```
+
+1. Create a `.dockerignore` file to ensure that any files related to a local
+   build do not affect the container that you build for deployment.
+
+   ```ignore
+   Dockerfile
+   README.md
+   *.pyc
+   *.pyo
+   *.pyd
+   __pycache__
+   ```
+
+**NOTE:** Use Docker to build the sample code into a container. To build and
+push with Docker Hub, run these commands replacing `{username}` with your Docker
+Hub username.
+
+1. Use Docker to build the sample code into a container, then push the container
+   to the Docker registry:
+
+   ```shell
+   # Build the container on your local machine
+   docker build -t {username}/helloworld-python .
+
+   # Push the container to docker registry
+   docker push {username}/helloworld-python
+   ```
+
+## Deploying the app
+
+1. After the build has completed and the container is pushed to Docker Hub, you
+   can deploy the app into your cluster.
+
+   {{< tabs name="helloworld_python" default="kn" >}} {{% tab name="yaml" %}}
+
+   1. Create a new file, `service.yaml` and copy the following service
+      definition into the file. Make sure to replace `{username}` with your
+      Docker Hub username.
+
+      ```yaml
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      metadata:
+        name: helloworld-python
+        namespace: default
+      spec:
+        template:
+          spec:
+            containers:
+              - image: docker.io/{username}/helloworld-python
+                env:
+                  - name: TARGET
+                    value: "Python Sample v1"
+      ```
+
+   Ensure that the container image value in `service.yaml` matches the container
+   you built in the previous step. Apply the configuration using `kubectl`:
+
+   ```shell
+   kubectl apply --filename service.yaml
+   ```
+
+   {{< /tab >}} {{% tab name="kn" %}}
+
+   With `kn` you can deploy the service with
+
+   ```shell
+   kn service create helloworld-python --image=docker.io/{username}/helloworld-python --env TARGET="Python Sample v1"
+   ```
+
+   This will wait until your service is deployed and ready, and ultimately it
+   will print the URL through which you can access the service.
+
+   {{< /tab >}} {{< /tabs >}}
+
+   During the creation of your service, Knative performs the following steps:
+
+   - Create a new immutable revision for this version of the app.
+   - Network programming to create a route, ingress, service, and load balance
+     for your app.
+   - Automatically scale your pods up and down, including scaling down to zero
+     active pods.
+
+## Verification
+
+1. Run one of the followings commands to find the domain URL for your service.
```

Comment:
Do we need to remark that xip.io address will be only if the default domain job is running alongside?

Reply:
not sure, as we have marked this as example.

Reply:
well still... people might expect xip.io and it won't be there

Reply:
@vagababov added a note for this.